### ISSUE TITLE
Dynamic relation fields

### ISSUE THREAD

**Issue by RuudvdvNB** [pimcore/pimcore#5969]
When building a PIM for our clients the question is raised that they want to be able to filter specific relational fields depending on other relation fields in the same object.

For instance, when you create a new object the client sets a `mainCategory` which is a one-to-many relation. There is a second field `subCategory`, but when you're using the search it returns all the subCategory objects that are found. In this case i only want to be able to return the subCategories that are related to the selected `mainCategory`. 

Is something like this possible or would this be a new feature?

**Comment by dpfaffenbauer** [comment]
Let the customer only select the sub-category and automatically add the main-category based on that ;)

But, I get your point. There are many scenarios where a feature like that would be cool. Guess that could be solved with something similar to a many-to-many or many-to-one relation field with a configured service to filter the possible selections.

**Comment by RuudvdvNB** [comment]
Yeah, the category was just an example because i couldn't come up with a better one that was easy to explain ;)
